src/main/java/BottleOpener/Event.java

```diff
@@ -0,0 +1,161 @@
+                .appendOptional(DateTimeFormatter.ofPattern("MM/dd/yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM/dd/yyyy"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM-dd-yyyy HHmm"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM-dd-yyyy HH:mm"))
+                .appendOptional(DateTimeFormatter.ofPattern("MM-dd-yyyy"))
+                .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+                .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+                .toFormatter();
+    }
+
+    private static final DateTimeFormatter OUT_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    public LocalDateTime start;
+    public LocalDateTime end;
+
+    /**
+     * Constructs a new BottleOpener.Event task with the specified description, start, and end times.
+     * The task is initially marked as not done.
+     * <p>
+     * The start and end times are parsed from the given strings using the defined {@link #FORMATTER}.
+     * </p>
+     *
+     * @param description The description of the BottleOpener.Event task.
+     * @param start       The start time of the event as a string.
+     * @param end         The end time of the event as a string.
+     */
+    public Event(String description, String start, String end) {
+        super(description);
+        try {
+            this.start = LocalDateTime.parse(start, FORMATTER);
+            this.end = LocalDateTime.parse(end, FORMATTER);
+        } catch (DateTimeParseException e) {
+            System.out.println("Invalid event format!");
+            this.start = LocalDateTime.now();
+            this.end = LocalDateTime.now();
+        }
+    }
+
+    /**
+     * Constructs a new BottleOpener.Event task with the specified description, status, start, and end times.
+     * <p>
+     * The start and end times are parsed from the given strings using the defined {@link #FORMATTER}.
+     * If the times are not in a valid format, the current date and time will be used.
+     * </p>
+     *
+     * @param description The description of the BottleOpener.Event task.
+     * @param status      The completion status of the task; {@code true} if the task is done, {@code false} otherwise.
+     * @param start       The start time of the event as a string.
+     * @param end         The end time of the event as a string.
+     */
+    public Event(String description, boolean status, String start, String end) {
+        super(description, status);
+        try {
+            this.start = LocalDateTime.parse(start, FORMATTER);
+            this.end = LocalDateTime.parse(end, FORMATTER);
+        } catch (DateTimeParseException e) {
+            System.out.println("Invalid event format!");
+            this.start = LocalDateTime.now();
+            this.end = LocalDateTime.now();
+        }
```

Comment:
nit: does not seem right to force the event to have current time for start and end if the format is wrong